python/pyiceberg/catalog/rest.py

```diff
@@ -200,32 +196,36 @@ def _split_credential(token: str) -> Tuple[str, str]:
 
     @property
     def headers(self) -> Properties:
-        return {
-            AUTHORIZATION_HEADER: f"{BEARER_PREFIX} {self.token}",
+        headers = {
             "Content-type": "application/json",
             "X-Client-Version": __version__,
         }
+        if self.token:
+            headers[AUTHORIZATION_HEADER] = f"{BEARER_PREFIX} {self.token}"
+        return headers
 
-    def url(self, endpoint: str, prefixed: bool = True, **kwargs) -> str:
+    def url(self, uri: str, prefixed: bool = True, **kwargs) -> str:
         """Constructs the endpoint
 
         Args:
+            uri: Resource identifier that points to the REST catalog
             prefixed: If the prefix return by the config needs to be appended
 
         Returns:
             The base url of the rest catalog
         """
 
-        url = self.host
+        url = self.uri
         url = url + "v1/" if url.endswith("/") else url + "/v1/"
 
         if prefixed:
             url += self.config.get(PREFIX, "")
             url = url if url.endswith("/") else url + "/"
 
-        return url + endpoint.format(**kwargs)
+        return url + uri.format(**kwargs)
```

Comment:
Looks like this refactor caught a little too much. Both the `endpoint` argument and `url` instance field have been changed to `uri`. I think you want to revert the `endpoint` to `uri` rename?

I think this still works since `self.uri` and `uri` are distinct, but it makes this method harder to read.

Reply:
Fair point, I've reverted it 👍🏻